src/Factory/ProductDocumentFactory.php

```diff
@@ -70,115 +67,121 @@ public function __construct(
         $this->assertClassExtends($priceDocumentClass, PriceDocument::class);
         $this->priceDocumentClass = $priceDocumentClass;
 
+        $this->assertClassExtends($productTaxonDocumentClass, ProductTaxonDocument::class);
+        $this->productTaxonDocumentClass = $productTaxonDocumentClass;
+
         $this->assertClassExtends($taxonDocumentClass, TaxonDocument::class);
         $this->taxonDocumentClass = $taxonDocumentClass;
 
         $this->attributeWhitelist = $attributeWhitelist;
     }
 
     /**
-     * @param ProductInterface $syliusProduct
+     * @param ProductInterface $product
      * @param LocaleInterface $locale
      * @param ChannelInterface $channel
      *
      * @return ProductDocument
      */
-    public function createFromSyliusSimpleProductModel(ProductInterface $syliusProduct, LocaleInterface $locale, ChannelInterface $channel)
+    public function createFromSyliusSimpleProductModel(ProductInterface $product, LocaleInterface $locale, ChannelInterface $channel)
```

Comment:
I've done a little refactoring there to follow the class names, `$product` is a Sylius entity, `$productDocument` is its ES representation, `$productView` is its view representation etc.